src/DepthImageToLaserScanROS.cpp

```diff
@@ -99,7 +92,7 @@ void DepthImageToLaserScanROS::depthCb(const sensor_msgs::msg::Image::SharedPtr
   try
   {
     sensor_msgs::msg::LaserScan::SharedPtr scan_msg = dtl_.convert_msg(image, cam_info_);
-    scan_pub_->publish(scan_msg);
+    scan_pub_->publish(*scan_msg);
```

Comment:
To make this compatible with zero-copy intra-process comms, we should change `convert_msg` to return a `std::unique_ptr`, and then publish by doing:

```
scan_pub_->publish(std::move(scan_msg));
```